imod/mf6/utilities/imod5_converter.py

```diff
@@ -26,3 +28,20 @@ def convert_ibound_to_idomain(
     )
     # Fill the remaining nans at tops and bottoms with 0
     return idomain_float.fillna(0).astype(int)
+
+
+def fill_missing_layers(
+    source: xr.DataArray, full: xr.DataArray, fillvalue: Union[float | int]
+) -> xr.DataArray:
+    """
+    This function takes a source grid in which the layer dimension is
+    incomplete. It creates a result-grid which has the same layers as the "full"
+    grid, which is assumed to have all layers. The result has the values in the
+    source for the layers that are in the source. For the other layers, the
+    fillvalue is assigned.
+    """
+    result = zeros_like(full)
```

Comment:
I'm pretty confident xarray has tooling for this, which ideally also supports out-of-memory (unlike calling `values`).

Maybe `align`?

```python
import xarray as xr
da1 = xr.DataArray([1, 2, 4], coords={"x": [1, 2, 4]}, dims=["x"])
da2 = xr.DataArray([0, 0, 0, 0, 0], coords={"x": [1, 2, 3, 4, 5]}, dims=["x"])
aligned, _ = xr.align(da1, da2, join="right")
```

https://docs.xarray.dev/en/stable/generated/[xarray.align](https://docs.xarray.dev/en/stable/generated/xarray.align.html).html

Reply:
This was a previous comment of me as well. I had a code suggestion using reindex, but it appears to be ignored or not pushed yet?

Reply:
@JoerivanEngelen  Indeed, I overlooked it, Now it is implemented, with reindexing